ams/ams-server/src/main/java/com/netease/arctic/ams/server/optimize/BaseArcticOptimizePlan.java

```diff
@@ -183,12 +183,11 @@ protected BaseOptimizeTask buildOptimizeTask(@Nullable List<DataTreeNode> source
   }
 
   public boolean baseTableCacheAll() {
-    Snapshot snapshot;
     if (arcticTable.isKeyedTable()) {
-      snapshot = arcticTable.asKeyedTable().baseTable().currentSnapshot();
-      if (snapshot != null && !snapshotIsCached.test(snapshot.snapshotId())) {
+      long snapshotId = UnKeyedTableUtil.getSnapshotId(arcticTable.asKeyedTable().baseTable());
```

Comment:
The `baseTable` is refreshed twice in `tableNeedPlan()` and `baseTableCacheAll()`

Reply:
OK, I move it to baseTableCacheAll()

Reply:
How about refreshing both `BaseStore` and `ChangeStore` first in `tableNeedPlan` method,  
This will make the code easier to understand.

Reply:
OK